subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc

```diff
@@ -325,11 +326,14 @@ You should check for existence of optional project properties before you access
 
 You can adjust JVM options for Gradle in the following ways:
 
-The `org.gradle.jvmargs` Gradle property controls the VM running the build. It defaults to `-Xmx512m "-XX:MaxMetaspaceSize=256m"`
+The `org.gradle.jvmargs` Gradle property specifies arguments for the JVM running the build. The default values are `-Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`.
 
-.Changing JVM settings for the build VM
+JVM arguments provided via `org.gradle.jvmargs` are combined with the defaults, with special handling for `-Xms` and `-Xmx`: setting either these values will remove the default for the other.
+This avoids the case where a default value is incompatible with a user-supplied value, such as having a minimum memory setting higher than the maximum.
```

Comment:
```suggestion
Gradle combines JVM arguments provided via `org.gradle.jvmargs` with the default JVM arguments. Setting either `-Xms` and `-Xmx` removes the default for the other.
This avoids incompatible combinations of default and user-supplied values, such as a minimum memory setting higher than the maximum.
```
Suggestion: active voice rephrase to make the first sentence more readable. Trimmed some words from the second sentence for the same reason.